controller/scheduler-v2/scheduler.go

```diff
@@ -390,6 +525,66 @@ func (s *Scheduler) Jobs() map[string]*Job {
 	return jobs
 }
 
+// Channel handlers
+type chanHandler struct {
+	channel chan interface{}
+	handler func(interface{}) error
+}
+
+func (s *Scheduler) jobRequestHandler(i interface{}) error {
+	req, ok := i.(*JobRequest)
+	if !ok {
+		return fmt.Errorf("Failed to cast to JobRequest")
+	}
+	return s.HandleJobRequest(req)
+}
+
+func (s *Scheduler) rectifyJobsHandler(i interface{}) error {
+	return s.RectifyJobs()
+}
+
+func (s *Scheduler) formationChangeHandler(i interface{}) error {
+	fc, ok := i.(*ct.ExpandedFormation)
+	if !ok {
+		return fmt.Errorf("Failed to cast to ExpandedFormation")
+	}
+	return s.FormationChange(fc)
+}
+
+func (s *Scheduler) formationSyncHandler(i interface{}) error {
+	return s.SyncFormations()
+}
+
+func (s *Scheduler) jobSyncHandler(i interface{}) error {
+	return s.SyncJobs()
+}
+
+func (s *Scheduler) handleChan(chHandler *chanHandler) bool {
+	log := s.log.New("fn", "handleChan")
+
+	select {
+	case test := <-chHandler.channel:
+		if chHandler.handler != nil {
+			if err := chHandler.handler(test); err != nil {
+				log.Error("error performing task", "err", err)
+			}
+		}
+		return true
+	default:
+		return false
+	}
+}
+
+func drainChannel(ch chan interface{}) {
+	for {
+		select {
+		case <-ch:
+		default:
+			return
+		}
+	}
+}
```

Comment:
Rather than draining channels, we should use channels with a buffer length of 1 and do non-blocking writes like:

```go
func triggerJobSync() {
	select {
	case jobSync <- struct{}{}
	default:
	}
}
```